### PR TITLE
Add better support for parameterised queries in @BigQueryType.fromQuery

### PR DESCRIPTION
**Pull request by regadas**
Fixes: #1078

### CODE REVIEW

File: scio-bigquery/src/it/scala/com/spotify/scio/bigquery/types/BigQueryTypeIT.scala [hunk 1]
```diff
@@ -37,13 +37,21 @@ object BigQueryTypeIT {
...
   @BigQueryType.fromQuery(
-    "SELECT word, word_count FROM `data-integration-test.partition_a.table_%s`", "$LATEST")
+    "SELECT word, word_count FROM `data-integration-test.partition_a.table_%s`",
+    "$LATEST")
   class SqlLatestT
 
+  @BigQueryType.fromQuery(
+    "SELECT word, word_count FROM `data-integration-test.partition_a.table_%s` LIMIT %d",
+    "$LATEST",
+    1)
```
Comment (jto) [inline scio-bigquery/src/it/scala/com/spotify/scio/bigquery/types/BigQueryTypeIT.scala]: Would be nice to have a test to check that
```scala
 @BigQueryType.fromQuery(
    "SELECT word, word_count FROM `data-integration-test.partition_a.table_%s` LIMIT %d",
    "$LATEST",
    "1")
```

would not compile

**Comment by jto** [comment]
Looks really cool 👍

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/TypeProvider.scala [hunk 2]
```diff
@@ -36,25 +36,28 @@ import com.spotify.scio.bigquery.{
...
   private[this] val logger = LoggerFactory.getLogger(this.getClass)
   private lazy val bigquery: BigQueryClient = BigQueryClient.defaultInstance()
+  private[this] val FormatSpecifierRegex =
+    "(%(\\d+\\$)?([-#+ 0,(\\<]*)?(\\d+)?(\\.\\d+)?([tT])?([a-zA-Z%]))".r
 
   def tableImpl(c: blackbox.Context)(annottees: c.Expr[Any]*): c.Expr[Any] = {
     import c.universe._
 
     val args = extractStrings(c, "Missing table specification")
+    val query = args.head.asInstanceOf[String]
```
Comment (regadas) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/TypeProvider.scala]: yeah this thing can be clean up as well

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/TypeProvider.scala [hunk 3]
```diff
@@ -74,24 +77,49 @@ private[types] object TypeProvider {
   }
 
   def schemaImpl(c: blackbox.Context)(annottees: c.Expr[Any]*): c.Expr[Any] = {
-    val schemaString = extractStrings(c, "Missing schema").head
+    val schemaString = extractStrings(c, "Missing schema").head.asInstanceOf[String]
```
Comment (regadas) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/TypeProvider.scala]: cleanup here as well

**Pull request merged**